[IMP] account: display moves to check for all journals on dashboard

william-andre opened the pull request:
Task 1966334

Purpose

We can create journal entries “to check” in all journal. If there are entries “to check” I want to see them from the dashboard.
  See: https://drive.google.com/a/odoo.com/file/d/1BioSLemU29eF_n3pBI4fHy4CV5811e-C/view?usp=drivesdk

Functional Spec

Do it exactly As the same as of the bank journals but open the “journal entries” list view with context “to check” activated
  See: https://drive.google.com/a/odoo.com/file/d/1hsHbgaZYa3_g65vPpP9Q9z2jx6cIFa8O/view?usp=drivesdk

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

sswapnesh commented:
Hello @william-andre 
I just tested this on Runbot Build http://497420-master-to-check-for-all-journals-6dc208.runbot21.odoo.com/web?debug#action=550&model=account.journal&view_type=kanban&menu_id=339 and it seems some Issue over there.
![Peek 2019-04-10 21-54](https://user-images.githubusercontent.com/16624719/55896189-598eb180-5bdb-11e9-92a6-b461a04452b4.gif)

I believe it should open entries in list view (As it does on Vendor Bill Journal)

Ps: Maybe better to have string like this `%s moves to check `--> `%s Move(s) To check`

sswapnesh commented:
Another one


Remove filter

![](https://image.prntscr.com/image/xg4UndQ1SWOj30QmKHPHAw.png)


![Screenshot from 2019-04-10 22-07-33](https://user-images.githubusercontent.com/16624719/55896937-23523180-5bdd-11e9-9a15-30f4d6a3b222.png)

sswapnesh commented on addons/account/models/account_journal_dashboard.py:
```diff
@@ -215,8 +214,12 @@ def get_journal_dashboard_datas(self):
             (number_waiting, sum_waiting) = self._count_results_and_sum_amounts(query_results_to_pay, currency, curr_cache=curr_cache)
             (number_draft, sum_draft) = self._count_results_and_sum_amounts(query_results_drafts, currency, curr_cache=curr_cache)
             (number_late, sum_late) = self._count_results_and_sum_amounts(late_query_results, currency, curr_cache=curr_cache)
+            lines_to_check_ids = self.env['account.move.line'].search([('journal_id', '=', self.id), ('move_id.to_check', '=', True)])
```
Can we add new stored related field (to_check) in account.move.line (move_id.to_check) as this will be called quite often.
(We already did this for journal_id at https://github.com/odoo/odoo/blob/12.0/addons/account/models/account_move.py#L638)

william-andre commented:
@sswapnesh Hello, I corrected the traceback you got. About your other point, we want the action to be different on the bank journals and the other journals. I changed the label on the bank journals to avoid confusion.

sswapnesh commented:
and Can I have your suggestion on adding new stored related field ?

william-andre commented:
@sswapnesh I added the related field too, following your suggestion

sswapnesh commented on addons/account/models/account_move.py:
```diff
@@ -668,6 +668,7 @@ def _compute_parent_state(self):
     date_maturity = fields.Date(string='Due date', index=True, required=True, copy=False,
         help="This field is used for payable and receivable journal entries. You can put the limit date for the payment of this line.")
     date = fields.Date(related='move_id.date', string='Date', index=True, store=True, copy=False, readonly=False)  # related is required
+    to_check = fields.Boolean(related='move_id.to_check', store=True, readonly=True)
```
Related fields are readonly by default.

william-andre replied:
I know

william-andre commented:
@qdp-odoo Wow I'm dumb

qdp-odoo commented on addons/account/models/account_journal_dashboard.py:
```diff
@@ -215,6 +215,9 @@ def get_journal_dashboard_datas(self):
             (number_late, sum_late) = self._count_results_and_sum_amounts(late_query_results, currency, curr_cache=curr_cache)
+            to_check_ids = self.env['account.move'].search([('journal_id', '=', self.id), ('to_check', '=', True)])
+            number_to_check = len(to_check_ids)
+            to_check_balance = sum([r.amount for r in to_check_ids])
```
use search_read() instead, as the dashboard may compute a lot of information it is important to be aware of potential performances issues

william-andre replied:
I also tried to do it for the bank journals, but its trickier.

qdp-odoo commented:
@robodoo r+

robodoo commented:
Merged, thanks!
